azurestack/resource_arm_virtual_network_gateway.go

```diff
@@ -0,0 +1,776 @@
+
+	peeringAddress := bgp["peering_address"].(string)
+	peerWeight := int32(bgp["peer_weight"].(int))
+
+	return &network.BgpSettings{
+		Asn:               utils.Int64(int64(bgp["asn"].(int))),
+		BgpPeeringAddress: &peeringAddress,
+		PeerWeight:        &peerWeight,
+	}
+}
+
+func expandArmVirtualNetworkGatewayIPConfigurations(d *schema.ResourceData) *[]network.VirtualNetworkGatewayIPConfiguration {
+	configs := d.Get("ip_configuration").([]interface{})
+	ipConfigs := make([]network.VirtualNetworkGatewayIPConfiguration, 0, len(configs))
+
+	for _, c := range configs {
+		conf := c.(map[string]interface{})
+
+		name := conf["name"].(string)
+		privateIPAllocMethod := network.IPAllocationMethod(conf["private_ip_address_allocation"].(string))
+
+		props := &network.VirtualNetworkGatewayIPConfigurationPropertiesFormat{
+			PrivateIPAllocationMethod: privateIPAllocMethod,
+		}
+
+		if subnetID := conf["subnet_id"].(string); subnetID != "" {
+			props.Subnet = &network.SubResource{
+				ID: &subnetID,
+			}
+		}
+
+		if publicIP := conf["public_ip_address_id"].(string); publicIP != "" {
+			props.PublicIPAddress = &network.SubResource{
+				ID: &publicIP,
+			}
+		}
+
+		ipConfig := network.VirtualNetworkGatewayIPConfiguration{
+			Name: &name,
+			VirtualNetworkGatewayIPConfigurationPropertiesFormat: props,
+		}
+
+		ipConfigs = append(ipConfigs, ipConfig)
+	}
+
+	return &ipConfigs
+}
+
+func expandArmVirtualNetworkGatewayVpnClientConfig(d *schema.ResourceData) *network.VpnClientConfiguration {
+	configSets := d.Get("vpn_client_configuration").([]interface{})
+	conf := configSets[0].(map[string]interface{})
+
+	confAddresses := conf["address_space"].([]interface{})
+	addresses := make([]string, 0, len(confAddresses))
+	for _, addr := range confAddresses {
+		addresses = append(addresses, addr.(string))
+	}
+
+	var rootCerts []network.VpnClientRootCertificate
+	for _, rootCertSet := range conf["root_certificate"].(*schema.Set).List() {
+		rootCert := rootCertSet.(map[string]interface{})
+		name := rootCert["name"].(string)
+		publicCertData := rootCert["public_cert_data"].(string)
+		r := network.VpnClientRootCertificate{
+			Name: &name,
+			VpnClientRootCertificatePropertiesFormat: &network.VpnClientRootCertificatePropertiesFormat{
+				PublicCertData: &publicCertData,
+			},
+		}
+		rootCerts = append(rootCerts, r)
+	}
+
+	var revokedCerts []network.VpnClientRevokedCertificate
+	for _, revokedCertSet := range conf["revoked_certificate"].(*schema.Set).List() {
+		revokedCert := revokedCertSet.(map[string]interface{})
+		name := revokedCert["name"].(string)
+		thumbprint := revokedCert["thumbprint"].(string)
+		r := network.VpnClientRevokedCertificate{
+			Name: &name,
+			VpnClientRevokedCertificatePropertiesFormat: &network.VpnClientRevokedCertificatePropertiesFormat{
+				Thumbprint: &thumbprint,
+			},
+		}
+		revokedCerts = append(revokedCerts, r)
+	}
+
+	// var vpnClientProtocols []network.VpnClientProtocol
```

Comment:
Could we get a comment about why these are commented out? not supported in azure stack API version ####-##-## would be ideal.